src/DistributedArrays.jl

```diff
@@ -880,13 +880,37 @@ for f in (:+, :-, :div, :mod, :rem, :&, :|, :$)
             B = samedist(A, B)
             map_localparts($f, A, B)
         end
+        function ($f){T}(A::DArray{T}, B::Array{T})
+            size(A) == size(B) || throw(DimensionMismatch("first array has size $(size(A)) but second array has $(size(B))"))
+            return DArray(A) do I
+                ($f)(localpart(A), B[I...])
```

Comment:
Hmmm, renaming should not be an issue.

I don't see an obvious way to send only the required part of B in the init function. Another option is to implement a generic constructor of the type (Function, Union{DArray, Array}, Union{Array, DArray},) which optimizes accordingly.